ios/UARCTModule/UARCTMessageViewManager.m

```diff
@@ -0,0 +1,23 @@
+/* Copyright Airship and Contributors */
+
+#import "UARCTMessageViewManager.h"
+#import "UARCTMessageView.h"
+
+#import <React/RCTBridge.h>
+#import <React/RCTUIManager.h>
+
+
+@implementation UARCTMessageViewManager
+
+RCT_EXPORT_VIEW_PROPERTY(onLoadStarted, RCTDirectEventBlock)
+RCT_EXPORT_VIEW_PROPERTY(onLoadFinished, RCTDirectEventBlock)
+RCT_EXPORT_VIEW_PROPERTY(onLoadError, RCTDirectEventBlock)
+RCT_EXPORT_VIEW_PROPERTY(onClose, RCTDirectEventBlock)
+RCT_REMAP_VIEW_PROPERTY(messageId, messageID, NSString)
+RCT_EXPORT_MODULE(UARCTMessageView)
+
+- (UIView *)view {
```

Comment:
I remember this being a problem when I looked into this eons ago. How does this work? Like where does the view live once you've created it, and what view controller does it attach to?

Reply:
The docs did not go into much detail here. The ViewManager is singleton and this is called for every `UAMessageView` instance in the react view layout. The view will have its frame  overriden, so you can construct it with CGRectZero.